### PR TITLE
[LMLayer] Pass searchTermToKey function to Trie model

### PR DESCRIPTION
**Pull request by eddieantonio**
Adds the `searchTermToKey()` function to the Trie model, allowing models to define a custom way to simplify or "dumbify" search terms. This includes things like changing every thing to lowercase, removing accents, etc.

This changes:
 - [x] The constructor of the Trie model in the worker
 - [x] The compiler, to produce code that passes the `searchTermToKey()` function

### CODE REVIEW

**Comment by keyman-server** [comment]
[OSK renders for Windows](https://build.palaso.org/repository/download/KeymanDesktop_TestPrRenderOnScreenKeyboards/152103:id/kvk/index.html)

**Comment by eddieantonio** [comment]
Lexical models can now provide their own search key simplification functions!

Given a model that looks like this:

```typescript
(new LexicalModelCompiler).compile({
    format: 'trie-2.0',
    sources: ['aaa.tsv'],
    searchTermToKey: function(text) {
        return text.toLowerCase().replace(/[^a]/g, '');
    },
});
```

...and a word list† like this:

|Word	| Count |
|---------|---|
|aardvark	| 1 |
|artichoke	| 2 |
|armoranth	| 5 |
|Ann Arbour	| 10 |


The compiler will generate code like this:

```javascript
(function() {
'use strict';
LMLayerWorker.loadModel(new models.TrieModel({"type":"internal","weight":10,"values":["a"],"children":{"a":{"type":"internal","weight":10,"values":["a","﷐"],"children":{"a":{"type":"leaf","weight":10,"entries":[{"key":"aa","weight":10,"content":"Ann Arbour"},{"key":"aa","weight":5,"content":"armoranth"},{"key":"aaa","weight":1,"content":"aardvark"}]},"﷐":{"type":"leaf","weight":2,"entries":[{"key":"a","weight":2,"content":"artichoke"}]}}}}}, {
  searchTermToKey: function(text) {
        return text.toLowerCase().replace(/[^a]/g, '');
    },
}));
})();
```

Try the above model using `common/test/predictive-text`!

---

†Yes, @jahorton, I managed to spell an American town's name in a Canadian way. I'm sorry 🍁

File: developer/js/index.ts [hunk 1]
```diff
@@ -232,6 +162,91 @@ export default class LexicalModelCompiler {
...
+        break;
+      case 'trie-2.0':
+        func += `LMLayerWorker.loadModel(new models.TrieModel(${
+          createTrieDataStructure(sources, modelSource.searchTermToKey)
+        }, {\n`;
+        if (wordBreakingSource) {
+          func += `  wordBreaking: ${wordBreakingSource},\n`;
+        }
+        if (modelSource.searchTermToKey) {
+          func += `  searchTermToKey: ${modelSource.searchTermToKey.toString()},\n`;
```
Comment (jahorton) [inline developer/js/index.ts]: I see we're assuming that `searchTermToKey` will be an inline-defined function within the model's base declaration?  I suppose that's fine as long as it's simple enough - I tested it out with a simple dummy function to ensure compilation worked.

Comment (eddieantonio) [reply developer/js/index.ts]: Yup! If the need arises to make this more complicated later, we can always augment to this definition.

**Pull request merged**